src/_nebari/provider/cloud/google_cloud.py

```diff
@@ -15,232 +19,144 @@ def check_credentials() -> None:
-    """Return a dict of available projects."""
+def load_credentials():
     check_credentials()
-    output = subprocess.check_output(
-        ["gcloud", "projects", "list", "--format=json(name,projectId)"]
-    )
-    data = json.loads(output)
-    return {_["name"]: _["projectId"] for _ in data}
+    credentials = os.environ["GOOGLE_CREDENTIALS"]
+    project_id = os.environ["PROJECT_ID"]
+
+    # Scopes need to be explicitly defined when using workload identity
+    # federation.
+    scopes = ["https://www.googleapis.com/auth/cloud-platform"]
+
+    # Google credentials are stored as strings in GHA secrets so we need
+    # to determine if the credentials are stored as a file or not before
+    # reading them
+    if credentials.endswith(".json"):
+        loaded_credentials, _ = load_credentials_from_file(credentials, scopes=scopes)
+    else:
+        loaded_credentials, _ = load_credentials_from_dict(
+            json.loads(credentials), scopes=scopes
+        )
+
+    return loaded_credentials, project_id
 
 
 @functools.lru_cache()
 def regions() -> Set[str]:
-    """Return a set of available regions."""
-    check_credentials()
-    output = subprocess.check_output(
-        ["gcloud", "compute", "regions", "list", "--format=json(name)"]
-    )
-    data = json.loads(output)
-    return {_["name"] for _ in data}
+    """Return a dict of available regions."""
+    credentials, project_id = load_credentials()
+    client = compute_v1.RegionsClient(credentials=credentials)
+    response = client.list(project=project_id)
 
-
-@functools.lru_cache()
-def zones(project: str, region: str) -> Dict[str, str]:
-    """Return a dict of available zones."""
-    check_credentials()
-    output = subprocess.check_output(
-        ["gcloud", "compute", "zones", "list", "--project", project, "--format=json"]
-    )
-    data = json.loads(output.decode("utf-8"))
-    return {_["description"]: _["name"] for _ in data if _["name"].startswith(region)}
+    return {region.name for region in response}
 
 
 @functools.lru_cache()
 def kubernetes_versions(region: str) -> List[str]:
     """Return list of available kubernetes supported by cloud provider. Sorted from oldest to latest."""
-    check_credentials()
-    output = subprocess.check_output(
-        [
-            "gcloud",
-            "container",
-            "get-server-config",
-            "--region",
-            region,
-            "--format=json",
-        ]
+    credentials, project_id = load_credentials()
+    client = container_v1.ClusterManagerClient(credentials=credentials)
+    response = client.get_server_config(
+        name=f"projects/{project_id}/locations/{region}"
     )
-    data = json.loads(output.decode("utf-8"))
-    supported_kubernetes_versions = sorted([_ for _ in data["validMasterVersions"]])
-    return filter_by_highest_supported_k8s_version(supported_kubernetes_versions)
-
-
-@functools.lru_cache()
-def instances(project: str) -> Dict[str, str]:
```

Comment:
This function would be useful to do the same thing as AWS and Digital Ocean do and check instance types are valid before deploying.  I'd be okay if you just open an issue to re-add that functionality at some point though.  Search the repo for `instances(` and see how its used in infrastructure's \_\_init\_\_.py file.

Reply:
@Adam-D-Lewis great catch. I just created [an issue](https://github.com/nebari-dev/nebari/issues/2718) to track that and make sure we add that functionality for Google Cloud.